src/libifconfig_internal.c

```diff
@@ -39,6 +40,19 @@
 #include "libifconfig_internal.h"
 
 int
+ifconfig_getifaddrs(ifconfig_handle_t *h)
+{
+	int ret;
+
+	if (h->ifap == NULL) {
+		ret = getifaddrs(&h->ifap);
+		/*h->ifa_cursor = h->ifap;*/
+		return (ret);
+	} else
```

Comment:
How about if I call freeifaddrs at the end of ifconfig_for_each_iface?  That would prevent applications from cacheing ifaddrs information for longer than it takes to iterate through each interface.